src/api/app/models/kiwi/image/xml_parser.rb

```diff
@@ -103,6 +106,22 @@ def description
           specification:    attributes['specification']
         )
       end
+
+      def preference_type
+        attributes =
+          if xml_hash['preferences'].present? && xml_hash['preferences']['type'] && xml_hash['preferences']['type']['containerconfig'].present?
+            xml_hash['preferences']['type']
+          end
```

Comment:
It might be worth to move this into a separate method. That would probably also silence the code climate warning:-)